python/node_similarity.py

```diff
@@ -0,0 +1,285 @@
+from enum import Enum
+from math import sqrt
+from typing import Set, List, Callable, Tuple, Union
+from itertools import product
+import mgp
+
+# not implemented for directed graphs
+# not implemented for multigraphs
+# not taking into account edge weights
+
+neighbors_dict = dict()
+
+
+class Mode(Enum):
+    """
+    Enum for mode parameter
+    """
+
+    CARTESIAN = "cartesian"
+    PAIRWISE = "pairwise"
+
+
+@mgp.read_proc
+def jaccard(
+    context: mgp.ProcCtx, node1: mgp.Any, node2: mgp.Any, mode: str = "cartesian"
+) -> mgp.Record(node1=mgp.Vertex, node2=mgp.Vertex, similarity=float):
+    """This procedure calls a method for calculating the similarity between two nodes (or two tuples of nodes)
+    so that the used node similarity measure is the Jaccard similarity.
+
+    :param node1: A node or a tuple of nodes
+    :type node1: mgp.Vertex, tuple
+    :param node2: A node or a tuple of nodes
+    :type node2: mgp.Vertex, tuple
+    :param mode: Can be `pairwise` for pairwise similarity calculation or `cartesian`
+    for calculating the similarity of the Cartesian product of given tuples.
+    The default value is set to be a Cartesian product.
+    :type mode: str, optional
+
+    :return: This procedure returns a mgp.Record with 3 fields:
+    the first node, the second node and the Jaccard similarity between them
+    :rtype: mgp.Record
+    """
+
+    return _calculate_similarity(node1, node2, _calculate_jaccard, Mode(mode))
```

Comment:
What happens if mode is something that is not in ["pairwise", "cartesian"]

Reply:
The enum throws an error. I fixed that so that the called function throws an error.